### PR TITLE
feat: add interchain-security

### PR DESCRIPTION
**Pull request by hoank101**
- Interchain v4 using sdk50, ibc8

## Summary by CodeRabbit

- **New Features**
	- Added interchain security modules (`ccv`) including `ccvconsumer`, `ccvdistr`, `ccvgov`, and `ccvstaking.
	- Implemented whitelisting functions for governance transactions.
	- Introduced a consumer section in the genesis file for enhanced testing capabilities.
- **Enhancements**
	- Improved `ante` handler with new `ConsumerKeeper` for better module interactions.
	- Enhanced local development experience with updated `start_local_node.sh` script for the "eve" network.
- **Bug Fixes**
	- Adjusted build commands and file paths in `Makefile` for better compatibility across different operating systems.
- **Documentation**
	- Added comments and documentation in newly introduced functions and modules for clarity.

### CODE REVIEW

**Comment by faddat** [comment]
Yoooo omg that was fast.

Great work!

File: scripts/start_local_node.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+rm -rf ${EVE_HOME}
+
+$EVED init eve-local --chain-id $CHAIN_ID
+
+sed -i -E "s|minimum-gas-prices = \".*\"|minimum-gas-prices = \"0${DENOM}\"|g" $app_toml
+sed -i -E '/\[api\]/,/^enable = .*$/ s/^enable = .*$/enable = true/' $app_toml
+
+sed -i -E "s|chain-id = \"\"|chain-id = \"${CHAIN_ID}\"|g" $client_toml
+sed -i -E "s|keyring-backend = \"os\"|keyring-backend = \"test\"|g" $client_toml
+sed -i -E "s|node = \".*\"|node = \"tcp://localhost:26657\"|g" $client_toml
```
[the remark this answers is not in the document]
Comment (faddat) [reply scripts/start_local_node.sh:30]: oh, interesting test approach.

**Comment by hoank101** [comment]
now it work ok, but need fix test ci

**Comment by faddat** [comment]
You know that this is the fastest I've ever seen anyone do something similar, ever?

**Pull request merged**